### DISCUSSION TITLE
Registering an implementation that supports more than one interface

### DISCUSSION THREAD

**Discussion by MartinRobins** [PrismLibrary/Prism#3007] [General] [answered]
I have an interface that is inherited from another...

```
public interface ICommonSettingsProvider {...}

public interface ISpecificSettingsProvider : ICommonSettingsProvider {...}
```

I then have an implementation...

`public class SpecificSettingsProvider : ISpecificSettingsProvider {...}`

Currently, I find that I have to register both interfaces in the container...

```
containerRegistry.RegisterSingleton<ISpecificSettingsProvider, SpecificSettingsProvider>();

ISpecificSettingsProvider specificSettingsProvider = ContainerLocator.Container.Resolve<SpecificSettingsProvider>();
containerRegistry.RegisterInstance<ICommonSettingsProvider>(specificSettingsProvider);
containerRegistry.RegisterInstance<ISpecificSettingsProvider>(specificSettingsProvider);
```

Is it possible for me to register this instance (or even just register as a singleton) the implementation for both interfaces at the same time or is this the best way to do it?

Thanks.

**Comment by thomasgalliker** [comment, marked as answer]
I think dryioc supports this with RegisterMany.

**Reply by MartinRobins** [reply]
@thomasgalliker That looks like the answer I was looking for, and it is supported by the Prism `IContainerRegistry` too. The single line...

`containerRegistry.RegisterManySingleton<SpecificSettingsProvider>(typeof(ICommonSettingsProvider), typeof(ISpecificSettingsProvider));`

... replaces all 4 of my original lines above.

Thanks,
